### ISSUE TITLE
Addons based on HTTP status code

### ISSUE THREAD

**Issue by humitos** [readthedocs/addons#385]
In https://github.com/readthedocs/addons/pull/373 we started making usage of the HTTP status code to decide whether or not show specific addons on those pages. We are not showing the flyout and ad on 404 pages, for example. However, if the page is customized (using `sphinx-notfound-page`) I think we still want to show them. What would be a good way to implement this logic? Should we add another META tag via CF to communicate if the 404 page is custom or not?

**Comment by humitos** [comment]
Examples:

- https://docs.godotengine.org/en/latest/page-not-found/
- https://sphinx-notfound-page.readthedocs.io/en/latest/not-found-page/

Those pages should show the flyout menu at the bottom right, but it's not there.

**Comment by agjohnson** [comment]
It doesn't feel like there is a good blanket default here. Even if a user has a custom 404 page they might still not want the flyout, notifications, or other addons on their error pages.

**Comment by humitos** [comment]
Yeah, that probably makes sense. What granularity of explicit configuration are you thinking about? Per addon? Per http status code? Both? Like, 

* Show flyout on 404 pages, but don't show notifications
* Show flyout on 404 pages but not on 403

**Comment by agjohnson** [comment]
Hrm, not sure really. I'm not feeling we need to offer configuration options by status code, that feels excessive.

If we have enough users asking for this, it maybe is worth a configuration option on each addon. I expect that themes are most likely to dictate whether error pages have our elements on them though. It would be good to design starting from there probably.

**Comment by humitos** [comment]
OK, that makes sense. I will leave this issue aside for now then until we have more feedback from users. There is no need to complicate it currently, in particular, if we don't have the use cases well defined.
